icecap-runtime-manager/src/main.rs

```diff
@@ -0,0 +1,201 @@
+//! IceCap-specific material for the Runtime Manager enclave
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE_MIT.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+#![feature(rustc_private)]
+#![no_main]
+#![feature(format_args_nl)]
+extern crate alloc;
+
+use bincode;
+use core::{convert::TryFrom, mem::size_of};
+use icecap_core::{
+    config::*,
+    logger::{DisplayMode, Level, Logger},
+    prelude::*,
+    ring_buffer::*,
+};
+use icecap_start_generic::declare_generic_main;
+use icecap_std_external;
+use runtime_manager::{
+    common_runtime::CommonRuntime,
+};
+use serde::{Deserialize, Serialize};
+
+mod icecap_runtime;
+
+declare_generic_main!(main);
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct Config {
+    event_nfn: Notification,
+    virtio_console_server_ring_buffer: UnmanagedRingBufferConfig,
+    badges: Badges,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct Badges {
+    virtio_console_server_ring_buffer: Badge,
+}
+
+fn  main(config: Config) -> Fallible<()> {
+    // TODO why do we need this?
+    icecap_runtime_init();
+
+    debug_println!("icecap-realmos: initializing...");
+
+    // enable ring buffer to serial-server
+    let virtio_console_client =
+        RingBuffer::unmanaged_from_config(&config.virtio_console_server_ring_buffer);
+    virtio_console_client.enable_notify_read();
+    virtio_console_client.enable_notify_write();
+    debug_println!("icecap-realmos: enabled ring buffer");
+
+    debug_println!("icecap-realmos: running...");
+    RuntimeManager::new(
+        virtio_console_client,
+        config.event_nfn,
+        config.badges.virtio_console_server_ring_buffer,
+    )
+    .run()
+}
+
+struct RuntimeManager {
+    channel: RingBuffer,
+    event: Notification,
+    virtio_console_server_ring_buffer_badge: Badge,
+    active: bool,
+}
+
+impl RuntimeManager {
+    fn new(
+        channel: RingBuffer,
+        event: Notification,
+        virtio_console_server_ring_buffer_badge: Badge,
+    ) -> Self {
+        Self {
+            channel: channel,
+            event: event,
+            virtio_console_server_ring_buffer_badge: virtio_console_server_ring_buffer_badge,
+            active: true,
+        }
+    }
+
+    fn run(&mut self) -> Fallible<()> {
+        let icecap_runtime = icecap_runtime::IcecapRuntime{};
+
+        println!("icecap_runtime_manager::linux_main accept run looping");
```

Comment:
I think this should use `debug_println!` and I think that `linux` bit is inaccurate.